### DISCUSSION TITLE
Out of date sub-dependency for @parcel/optimizer-htmlnano

### DISCUSSION THREAD

**Discussion by NovaAtWarren** [parcel-bundler/parcel#6302] [Q&A] [answered]
# What the problem is

When running `npm install`, it gives a warning that the following packages are out of date:

- **request-promise-native**
- **request**
- urix
- **har-validator**
- resolve-url

@parcel/optimizer-htmlnano depends on (directly and indirectly) the ones highlighted in bold.

# Why I think it's a problem

As the packages are deprecated, they may not receive security updates, and may (at some point) stop being available. It might also cause paranoid idiots such as myself to go on a wild goose chase, ruining their package.json, trying to work out what and where out of date packages are.

# What I think I need to do to fix it

1. Update the file `packages/optimizers/htmlnano/package.json` to include the most up-to-date version of htmlnano, and thus update all of it's dependencies to be in-date
2. Test to make sure I didn't break anything
3. PR to this repo

# What I'm not sure about

I'm not sure how to run tests or how to package it up into a bundle to be used in projects, without just downloading from NPM (which won't include any potential fixes I develop). How would you recommend I do this?

EDIT: Just looked further, it seems to be an issue with htmlnano requiring an old version of uncss? Still need to work out how to build and test, though.

ANOTHER EDIT: OK, looked again, it now looks like an issue with uncss using request. I feel a little silly now.

**Comment by mischnic** [comment, marked as answer]
Yes, unfortunately there is not much we can do. uncss and/or htmlnano have to update their dependencies.

The only warning we have control over is
```
warning parcel > @parcel/config-default > @parcel/resolver-default > @parcel/node-resolver-core > micromatch > snapdragon > source-map-resolve > resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
warning parcel > @parcel/config-default > @parcel/resolver-default > @parcel/node-resolver-core > micromatch > snapdragon > source-map-resolve > urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
```
but this wasn't fixed yet because the newest `micromatch` major version has slightly different behaviour.

**Reply by NovaAtWarren** [reply]
This makes sense now. I've made a PR on uncss to change from request to axios, and hopefully that'll get accepted, then brought into htmlnano.
Thanks for the help.
